File: CS/WpfApplication1/VieweModel.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;

namespace WpfApplication1
{
    public class ViewModel : IDisposable
    {
        public void Dispose()
        {
            if (_Data != null)
            {
                _Data.Dispose();
                _Data = null;
            }
        }
        public ViewModel()
        {

        }

        // Fields...
        private DataTable _Data;

        public DataTable Data
        {
            get
            {
                if (_Data == null)
                    _Data = CreateData(10);
                return _Data;
            }
            set
            {
                _Data = value;
            }
        }

        DataTable CreateData(int rows)
        {
            Random _r = new Random();

            DataTable dt = new DataTable();
            dt.Columns.Add("Ser", typeof(int));
            dt.Columns.Add("Arg", typeof(int));
            dt.Columns.Add("Value", typeof(int));
            for (int i = 0; i < rows; i++)
            {
                dt.Rows.Add(0, i, _r.Next(100));
                dt.Rows.Add(1, i, _r.Next(100));
            }
            return dt;
        }

    }
}

File: VB/WpfApplication1/VieweModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Linq

Namespace WpfApplication1
	Public Class ViewModel
		Implements IDisposable
		Public Sub Dispose() Implements IDisposable.Dispose
			If _Data IsNot Nothing Then
				_Data.Dispose()
				_Data = Nothing
			End If
		End Sub
		Public Sub New()

		End Sub

		' Fields...
		Private _Data As DataTable

		Public Property Data() As DataTable
			Get
				If _Data Is Nothing Then
					_Data = CreateData(10)
				End If
				Return _Data
			End Get
			Set(ByVal value As DataTable)
				_Data = value
			End Set
		End Property

		Private Function CreateData(ByVal rows As Integer) As DataTable
			Dim _r As New Random()

			Dim dt As New DataTable()
			dt.Columns.Add("Ser", GetType(Integer))
			dt.Columns.Add("Arg", GetType(Integer))
			dt.Columns.Add("Value", GetType(Integer))
			For i As Integer = 0 To rows - 1
				dt.Rows.Add(0, i, _r.Next(100))
				dt.Rows.Add(1, i, _r.Next(100))
			Next i
			Return dt
		End Function

	End Class
End Namespace
